### PR TITLE
Fixups for Build agent cleanup task

### PR DESCRIPTION
**Pull request by MattGal**
- Add MaximumTimeInMinutes, which will cause the build task to quit after the specified time (default 10.0 mins)
- Add MaximumWorkspacesToClean to prevent taking on every workspace on the machine
- Remove usage of Robocopy (which can fail to return)

@chcosta @smile21prc @weshaggard

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs [hunk 1]
```diff
@@ -338,6 +354,12 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
                 }
                 return true;
             }
+            catch (PathTooLongException)
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Are you just completely not trying to handle pathtoolongexception? Some repos, like core-setup, currently have long path issues

Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Correct.  I'd rather we drive them fixing their problem.

Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: As I am unable to contact you, I'm going to merge for now.  The reason I've given up on solving the path too long exceptions here is that it seems quite possible on a messed up machine for RoboCopy.exe to never return, leading to a build failure.  I'd rather take my chances with a single bad type of build filling the disk than guaranteed timeout and build failure from this.

Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Let's meet up and chat about this.  Haven't actually updated buildtools in CoreFX yet, but this has been the state of every timed-out-on-this-step build we've seen (waiting on RoboCopy to return for hours)

Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs]: Update: I'll put the robocopy stuff back in a new PR but with more logging and a timeout on the RoboCopy process.

**Pull request merged**